.github/workflows/build-and-test.yml

```diff
@@ -35,52 +35,79 @@ jobs:
           large-packages: true
           swap-storage: true
 
-      - name: Create k8s Kind Cluster
-        uses: helm/kind-action@v1.9.0
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build and export
+        uses: docker/build-push-action@v5
         with:
-          cluster_name: ${{ env.KIND_CLUSTER_NAME }}
-          config: test/kserve/kind_config.yaml
+          context: .
+          file: ./Dockerfile
+          tags: ${{ env.DEV_IMAGE }}
+          # outputs: type=oci,dest=/tmp/image.tar
+          outputs: type=docker,dest=/tmp/image.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: caikit-tgis-serving
+          path: /tmp/image.tar
+
+  compose-smoke-test:
+    name: Smoke Test
+    runs-on: ubuntu-latest
+
+    needs: build-image
 
-      - name: Install kserve
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Free Disk Space
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: caikit-tgis-serving
+          path: /tmp
+
+      - name: Load Docker image
         run: |
-          # Use the hack/quick_install.sh script from the kserve repo
-          curl https://raw.githubusercontent.com/opendatahub-io/kserve/v0.11.1.1/hack/quick_install.sh | bash
+          docker load --input /tmp/image.tar
 
-      - name: Patch Istio Ingressway
+      - name: Pull tgis image
         run: |
-          # bind the istio ingressway ports on the host so that we can access its services
-          kubectl patch deployments.apps -o yaml -n istio-system istio-ingressgateway -p '{
-            "spec":
-              {"template":
-                {"spec":
-                  {"containers":
-                    [
-                      {
-                        "name":
-                        "istio-proxy",
-                        "ports":
-                          [
-                            {
-                              "containerPort":8080,
-                              "hostPort":80
-                            },
-                            {
-                              "containerPort":8443,
-                              "hostPort":443
-                            }
-                          ]
-                      }
-                    ]
-                  }
-                }
-              }
-          }'
-
-      - name: Setup testing namespace
+          docker pull quay.io/opendatahub/text-generation-inference:stable
```

Comment:
Hey @dtrifiro, quick question, if we are testing a PR, how would we replace this image?

Reply:
Actually never mind, this is caikit-tgis, so tgis would always be stable.